# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformGtk.cpp
# Commit: c1bd0996764c
# Full Hash: c1bd0996764c49174c9169fe5550905ce5dcef88
# Author: Jed Davis <jld@mozilla.com>
# Date: 2021-07-06 21:42:42
# Regressor Bug: 1635451
# File Overlap Count: 1
# Description:
#   Bug 1635451 - Allow GLX to work in headless content processes. r=jgilbert
#   
#   This follows what we're already doing for EGL: a refcounted object
#   which can own the X connection, where we hold a weak reference from
#   the library object so that multiple contexts opportunistically share
# ==============================================================================

diff -r 2974b1ba2beb -r c1bd0996764c gfx/thebes/gfxPlatformGtk.cpp
--- a/gfx/thebes/gfxPlatformGtk.cpp	Tue Jul 06 07:42:41 2021 +0000
+++ b/gfx/thebes/gfxPlatformGtk.cpp	Tue Jul 06 07:42:42 2021 +0000
@@ -30,6 +30,7 @@
 #include "mozilla/FontPropertyTypes.h"
 #include "mozilla/gfx/2D.h"
 #include "mozilla/gfx/Logging.h"
+#include "mozilla/gfx/XlibDisplay.h"
 #include "mozilla/Monitor.h"
 #include "mozilla/Preferences.h"
 #include "mozilla/StaticPrefs_gfx.h"
@@ -686,8 +687,9 @@
         return;
       }
 
-      mGLContext = gl::GLContextGLX::CreateGLContext({}, mXDisplay, root,
-                                                     config, false, nullptr);
+      mGLContext = gl::GLContextGLX::CreateGLContext(
+          {}, gfx::XlibDisplay::Borrow(mXDisplay), root, config, false,
+          nullptr);
 
       if (!mGLContext) {
         lock.NotifyAll();
@@ -831,7 +833,7 @@
   }
 #  endif
 
-  // Only use GLX vsync when the OpenGL compositor / WebRedner is being used.
+  // Only use GLX vsync when the OpenGL compositor / WebRender is being used.
   // The extra cost of initializing a GLX context while blocking the main
   // thread is not worth it when using basic composition.
   //
@@ -847,7 +849,7 @@
 
     // Nvidia doesn't support GLX at the same time as EGL but Mesa does.
     if (!gfxVars::UseEGL() || (adapterDriverVendor.Find("mesa") != -1)) {
-      useGlxVsync = gl::sGLXLibrary.SupportsVideoSync();
+      useGlxVsync = gl::sGLXLibrary.SupportsVideoSync(DefaultXDisplay());
     }
     if (useGlxVsync) {
       RefPtr<VsyncSource> vsyncSource = new GtkVsyncSource();